### PR TITLE
IOSSimulator implementation.

### PR DESCRIPTION
**Pull request by iansf**
Also fixes some minor bugs with iOS and Android interactions.

### CODE REVIEW

File: lib/src/device.dart [hunk 1]
```diff
@@ -245,12 +259,239 @@ class IOSDevice extends _Device {
...
+  static const String className = 'IOSSimulator';
+  static final String defaultDeviceID = 'default_ios_sim_id';
+
+  static const String _macInstructions =
+      'To work with iOS devices, please install ideviceinstaller. '
+      'If you use homebrew, you can install it with '
+      '"\$ brew install ideviceinstaller".';
+
+  // static String _simulatorPath = path.join('/Applications', 'iOS Simulator.app',
+  //     'Contents', 'MacOS', 'iOS Simulator');
```
Comment (abarth) [inline lib/src/device.dart]: Commented out code?

File: lib/src/device.dart [hunk 2]
```diff
@@ -245,12 +259,239 @@ class IOSDevice extends _Device {
...
+
+  static String _xcrunPath = path.join('/usr', 'bin', 'xcrun');
+
+  // static String _plistBuddyPath = 'PlistBuddy';
```
Comment (abarth) [inline lib/src/device.dart]: ditto

**Comment by abarth** [comment]
LGTM

**Comment by iansf** [comment]
Now with tests that work (at least locally)...

**Pull request merged**